shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCXShuffleTransport.scala

```diff
@@ -414,55 +369,90 @@ class UCXShuffleTransport(shuffleServerId: BlockManagerId, rapidsConf: RapidsCon
         .setDaemon(true)
         .build))
 
+  // helper class to hold transfer requests that have a bounce buffer
+  // and should be ready to be handled by a `BufferReceiveState`
+  class PerClientReadyRequests(val bounceBuffer: BounceBuffer) {
+    val transferRequests = new ArrayBuffer[PendingTransferRequest]()
+    def addRequest(req: PendingTransferRequest): Unit = {
+      transferRequests.append(req)
+    }
+  }
+
   exec.execute(() => {
     while (inflightStarted) {
       try {
-        var perClientReq = mutable.Map[RapidsShuffleClient, BufferReceiveState]()
-        var removed = false
-        inflightMonitor.synchronized {
-          var head: PendingTransferRequest = altList.peek()
-          if (head == null) {
-            val waitRange = new NvtxRange("Transport throttling", NvtxColor.RED)
-            try {
-              inflightMonitor.wait(100)
-            } finally {
-              waitRange.close()
+        var req: PendingTransferRequest = null
+        val requestsToHandle = new ArrayBuffer[PendingTransferRequest]()
+        var continue = true
+        altList.synchronized {
+          while (inflightStarted && continue) {
+            req = altList.poll()
+            if (req == null && requestsToHandle.isEmpty) {
+              altList.wait(100)
+            } else if (req != null) {
+              requestsToHandle.append(req)
+            } else {
+              continue = false
             }
```

Comment:
I think this can be simplified a bit to remove the unnecessary `continue` variable and make it more clear the logic is "drain all current requests".  Something similar to this:
```suggestion
          req = altList.poll()
          while (inflightStarted && req == null) {
            altList.wait(100)
            req = altList.poll()
          }
          while (req != null) {
            requestsToHandle.append(req)
            req = altList.poll()

Reply:
Yes, agree that works well. Just testing this change.

Reply:
Ok included the patch in https://github.com/NVIDIA/spark-rapids/pull/994/commits/94eb3d80fee0b26346b207eff4b3717bbd057715